### ISSUE TITLE
[Menu.svelte] setOrigin fires before element has been bound

### ISSUE THREAD

**Issue by Bandit** [Brain-Bones/skeleton#172]
I think `setOrigin();` needs to be in `onMount` so it only fires after the `elemMenu` element has been bound

https://github.com/Brain-Bones/skeleton/blob/dev/src/lib/Menu/Menu.svelte#L30

`setOrigin` fires before `elemMenu` has been bound to the DOM element, so when `origin` is omitted or set to `auto` it leads to "undefined" as the `cOrigin` variable.

**Comment by Bandit** [comment]
Another option if you don't want to use `onMount` (which could lead to a FOUC) is to create a [Svelte Action](https://svelte.dev/docs#template-syntax-element-directives-use-action) inside this component and then use that, e.g. `use:setOrigin`?

**Comment by endigo9740** [comment]
Thanks for the heads up, I'll take a look!

**Comment by ssbm-oro** [comment]
I think I'm having an issue related to this. I just updated to the latest skeleton, and now if I have a Menu component in my `+layout.svelte`, I get this error:

```Uncaught (in promise) TypeError: Cannot read properties of null (reading 'addEventListener')
    at Menu.svelte:98:22
```

**Comment by endigo9740** [comment]
Can you share full full error message and provide snippet of the menu code please @ssbm-oro. This will make this easier to debug. Thanks!

**Comment by ssbm-oro** [comment]
Sorry, I thought I had copied the entire exception.

```
Menu.svelte:98 
        
       Uncaught (in promise) TypeError: Cannot read properties of null (reading 'addEventListener')
    at Menu.svelte:98:22
    at run (index.mjs:18:12)
    at Array.map (<anonymous>)
    at index.mjs:1831:45
    at flush (index.mjs:1075:17)
    at init (index.mjs:1923:9)
    at new Root (root.svelte? [sm]:19:25)
    at createProxiedComponent (svelte-hooks.js?v=d8a065e1:341:9)
    at new ProxyComponent (proxy.js?v=d8a065e1:242:7)
    at new Proxy<Root> (proxy.js?v=d8a065e1:349:11)
```

Here is my menu code in +layout.svelte, which if I replace with a plaintext placeholder, the error goes away:

```html
<Menu>
	<DiscordAvatar slot="trigger" user={user} size="md" hover={true}></DiscordAvatar>
	<Card slot="content">
		<Button variant="ring-accent" on:click={logout}>Logout</Button>
	</Card>
</Menu>
```

DiscordAvatar is just a simple wrapper around the Avatar component that just sets the correct src url.

It looks like the exception is being thrown from here:

https://github.com/Brain-Bones/skeleton/blob/dev/src/lib/Menu/Menu.svelte#L96

**Comment by ssbm-oro** [comment]
You've probably already figured this out, but I discovered that if I change the origin of the Menu to "tl" or something other than "auto", the error goes away. So it's specifically related to the auto-origin.

**Comment by endigo9740** [comment]
FYI I saw the additional details, thanks! I'm on the tail end of a few large updates, but once those wrap I'll turn my attention to these Menu issues. Thanks for your patience in the meantime @Bandit @ssbm-oro

**Comment by endigo9740** [comment]
FYI I've created a new PR addressing this here: https://github.com/Brain-Bones/skeleton/pull/178

I've opted for your original suggestion for now @Bandit, as I figure FOUC will only occur if the menu defaults open. This is probably an uncommon default for most projects, so I hope this can get us by for now.

I'm not really happy with how the Menu component operates in general, so I may look to revisit this prior to v1.0. No set ETA for that yet though.

FYI I couldn't replicate either error mentioned @ssbm-oro or @Bandit, but it may be the limitations of testing with the raw components versus when they're imported via a package. I'd recommend you test the change next release and notify me if the issue persists.

@thomasbjespersen This is ready for QA, see the PR above.
